src/symbolic_dimensions.jl

```diff
@@ -34,31 +40,47 @@ You can convert a quantity using `SymbolicDimensions` as its dimensions
 to one which uses `Dimensions` as its dimensions (i.e., base SI units)
 `uexpand`.
 """
-struct SymbolicDimensions{R} <: AbstractDimensions{R}
+struct SymbolicDimensions{R} <: AbstractSymbolicDimensions{R}
     nzdims::Vector{INDEX_TYPE}
     nzvals::Vector{R}
 end
 
-@inline dimension_names(::Type{<:SymbolicDimensions}) = ALL_SYMBOLS
+"""
+    SymbolicDimensionsSingleton{R} <: AbstractSymbolicDimensions{R}
+
+This special symbolic dimensions types stores a single unit or constant, and can
+be used for constructing symbolic units and constants without needing to allocate mutable storage.
+"""
+struct SymbolicDimensionsSingleton{R} <: AbstractSymbolicDimensions{R}
+    dim::INDEX_TYPE
+
+    SymbolicDimensionsSingleton(dim::INDEX_TYPE, ::Type{_R}) where {_R} = new{_R}(dim)
```

Comment:
I still think these type arguments are very non-idiomatic. The more standard approach would be just
```suggestion
    SymbolicDimensionsSingleton{R}(dim::INDEX_TYPE) where {R} = new{R}(dim)
```
(which is already the default)

Reply:
Added! Yeah I'm not sure why I had written that initially...